openupgrade_scripts/scripts/website/15.0.1.0/post-migration.py

```diff
@@ -1,6 +1,49 @@
+import re
+
 from openupgradelib import openupgrade
 
 
+def extract_footer_copyright_company_name(env):
+    """Replace Copyright content in the new v15 template so as not to lose
+    content from previous versions if it has been customised."""
+    footer_copyright_company_name_view = env["ir.ui.view"].search(
+        [
+            ("key", "=", "website.footer_copyright_company_name"),
+            ("website_id", "=", False),
+        ]
+    )
+    if not footer_copyright_company_name_view:
+        return
+    footer_copyright_company_name_arch = footer_copyright_company_name_view.arch_db
+    footer_copyright_company_name_pattern = (
+        r'<span class="o_footer_copyright_name mr-2">(.*?)<\/span>'
+    )
+    footer_copyright_company_name_matches = re.findall(
+        footer_copyright_company_name_pattern,
+        footer_copyright_company_name_arch,
+        re.DOTALL,
+    )
+    website_layout_views = env["ir.ui.view"].search(
+        [("key", "=", "website.layout"), ("website_id", "!=", False)]
+    )
+    for view in website_layout_views:
+        website_layout_arch = view.arch_db
+        website_layout_pattern = (
+            r'<span class="o_footer_copyright_name mr-2">(.*?)<\/span>'
+        )
+        website_layout_matches = re.findall(
```

Comment:
Footer section was embedded in main layout view with no inheritance?

Reply:
Before v15 yes
https://github.com/odoo/odoo/blob/053a6eb1e38cce57d7abe202c50bdc2f6798be81/addons/website/views/website_templates.xml#L178